sensors/movementsensor.go

```diff
@@ -0,0 +1,324 @@
+// Package sensors defines interfaces for sensors used by viam cartographer
+package sensors
+
+import (
+	"context"
+	"time"
+
+	"github.com/edaniels/golog"
+	"github.com/golang/geo/r3"
+	geo "github.com/kellydunn/golang-geo"
+	"github.com/pkg/errors"
+	"go.opencensus.io/trace"
+	"go.viam.com/rdk/components/movementsensor"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/spatialmath"
+	rdkutils "go.viam.com/rdk/utils"
+	"go.viam.com/rdk/utils/contextutils"
+)
+
+const (
+	replayTimeToleranceMsec           = 10 // Milliseconds
+	replayTimestampErrorMessage       = "replay sensor timestamp parse RFC3339Nano error"
+	timedMovementSensorReadingTimeout = 5 * time.Second
+)
+
+var (
+	defaultTime = time.Time{}
+	// ErrMovementSensorNeitherIMUNorOdometer denotes that the provided movement sensor does neither support
+	// an IMU nor a movement sensor.
+	ErrMovementSensorNeitherIMUNorOdometer = errors.New("'movement_sensor' must either support both LinearAcceleration and " +
+		"AngularVelocity, or both Position and Orientation")
+	// ErrNoValidReadingObtained denotes that the attempt to obtain a valid IMU or odometer reading failed.
+	ErrNoValidReadingObtained = errors.New("could not obtain a reading that satisfies the time tolerance requirement")
+)
+
+// TimedMovementSensor describes a sensor that reports the time the reading is from & whether or not it is
+// from a replay sensor.
+type TimedMovementSensor interface {
+	Name() string
+	DataFrequencyHz() int
+	TimedMovementSensorReading(ctx context.Context) (TimedMovementSensorReadingResponse, error)
+	Properties() MovementSensorProperties
+}
+
+// MovementSensorProperties contains information whether or not an IMU and/or odometer are supported.
+type MovementSensorProperties struct {
```

Comment:
Since we import movement sensor already, can we just use its properties' definition instead of redefining our own?